lib/shift_generator.rb

```diff
@@ -2,18 +2,19 @@
 require_relative 'offset_gen'
 
 class ShiftGenerator
-  attr_reader :key, :offset
+  attr_reader :key_gen, :offset_gen, :shift
 
   def initialize
-    @key = KeyGen.new
-    @offset = OffsetGen.new
+    @key_gen = KeyGen.new
+    @offset_gen = OffsetGen.new
```

Comment:
changed variable names here so that they could be distinguished between a key param and a key generator